drivers/sensor/ite/ite_tach_it51xxx/tach_ite_it51xxx.c

```diff
@@ -0,0 +1,227 @@
+ * 32.768k   +-----------+    |             |        | +-----------+   |
+ *                            |       +-----+-----+  | |   _   _   |<--+
+ *                            |------>|  T1L/MR   |<-+-|  | |_| |_ |<--+
+ *                            |       +-----------+    +-----------+   |
+ *                            |       capture pulses                  T1B (GPJ6)
+ *                            |       in sample rate
+ *                            |       period
+ *                            |
+ *                            |        Sample Rate = TACH_FREQ / 128
+ *                            |                      |
+ *                            |        Tachometer 2  |                T2A (GPJ0)
+ *                            |             |        | +-----------+   |
+ *                            |       +-----+-----+  | |   _   _   |<--+
+ *                            |------>|  T2L/MR   |<-+-|  | |_| |_ |<--+
+ *                            |       +-----------+    +-----------+   |
+ *                            |       capture pulses                  T2B (GPJ1)
+ *                            |       in sample rate
+ *                            |       period
+ *                            |
+ *
+ *
+ * Based on the counter value, we can compute the current RPM of external signal
+ * from encoders.
+ */
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/dt-bindings/sensor/it51xxx_tach.h>
+#include <errno.h>
+#include <soc.h>
+#include <soc_dt.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(tach_ite_it51xxx, CONFIG_SENSOR_LOG_LEVEL);
+
+#define TACH_FREQ IT51XXX_EC_FREQ
+
+/* 0xC0/0xD0/0xE0: Tach channel 0~2 achometer speed (2 bytes value) */
+#define REG_TACH_CH       0x00
+/* 0xC6/0xD6/0xE6: Tach channel 0~2 control 1 */
+#define REG_TACH_CH_CTRL1 0x06
+#define TACH_CH_DVS       BIT(1)
+#define TACH_CH_SEL       BIT(0)
+
+struct tach_it51xxx_config {
+	/* Tach channel register base address */
+	uintptr_t base;
+	/* Tachometer alternate configuration */
+	const struct pinctrl_dev_config *pcfg;
+	/* Select channel of tachometer */
+	int channel;
+	/* Number of pulses per round of tachometer's input */
+	int pulses_per_round;
+};
+
+struct tach_it51xxx_data {
+	/* Captured counts of tachometer */
+	uint16_t capture;
+};
+
+static bool tach_ch_is_valid(const struct device *dev, int tach_ch)
+{
+	const struct tach_it51xxx_config *const config = dev->config;
+	const uintptr_t base = config->base;
+	bool valid = false;
+	uint8_t reg_val;
+
+	if (tach_ch > IT51XXX_TACH_CHANNEL_B) {
+		LOG_ERR("Tach channel %d invalid, only support 0(A) or 1(B)", tach_ch);
+		return false;
+	}
+
+	reg_val = sys_read8(base + REG_TACH_CH_CTRL1);
+	if (((reg_val & TACH_CH_SEL) == tach_ch) && (reg_val & TACH_CH_DVS)) {
+		/* Tachometer channel match register setting and data valid */
+		valid = true;
```

Comment:
how about just `return true` here and `return false` at the end?

Reply:
Done